src/DotNetty.Transport/Channels/Sockets/TcpServerSocketChannel.cs

```diff
@@ -127,10 +127,13 @@ public TcpServerSocketChannelUnsafe(TcpServerSocketChannel channel)
 
             public override void FinishRead(SocketChannelAsyncOperation operation)
             {
-                Contract.Requires(this.channel.EventLoop.InEventLoop);
+                Contract.Assert(this.channel.EventLoop.InEventLoop);
 
                 TcpServerSocketChannel ch = this.Channel;
-                ch.ResetState(StateFlags.ReadScheduled);
+                if ((ch.ResetState(StateFlags.ReadScheduled) & StateFlags.Active) == 0)
```

Comment:
100% ch is in ReadScheduled state.

---
In reply to: [73931101](https://github.com/Azure/DotNetty/pull/152#discussion_r73931101) [](ancestors = 73931101)